model/dataset-filter/hierarchy/hierarchy.go

```diff
@@ -34,8 +34,9 @@ type Hierarchy struct {
 
 // AddAll ...
 type AddAll struct {
-	Amount string `json:"amount"`
-	URL    string `json:"url"`
+	Amount   string `json:"amount"`
+	URL      string `json:"url"`
+	ShowLink bool   `json:"show_link"`
```

Comment:
My 2 cents.

I am not a fan of the naming of these structs and properties, I understand these are names used elsewhere and haven't been added in this PR however think we can improve the one we have added.

e.g. from reading: `Data.AddAllFilters.ShowLink`
1. Data is not a very useful name; all variables store 'data' so it doesn't add any value. As the joke goes:
   

> Q: What’s the worst variable name ever?
> A: data
>
> Q: What’s the second-worst variable name?
> A: data2

2. `AddAll` is more of a verb than a noun, properties don't 'do'. Thus the name is misleading

3. `ShowLink` The name of this indicates the model is controlling what the view should render, it also indicates this property will only ever have a single use-case. This field is populated in the controller based on the question, 'does the hierarchy models children have data' So a name more like 'hierachyDataPresent' would be more accurate for what it represents; rather than naming it on how it is used (We may want to use it in other situations that don't involve showing a link) - Could be a better name than hierachyDataPresent this is just an example.